pkg/middleware/auth.go

```diff
@@ -97,6 +100,65 @@ func CanAdminPlugins(cfg *setting.Cfg, accessControl ac.AccessControl) func(c *c
 	}
 }
 
+func RoleAppPluginAuthAndSignedIn(accessControl ac.AccessControl, ps pluginstore.Store, features featuremgmt.FeatureToggles, logger log.Logger) func(c *contextmodel.ReqContext) {
+	return func(c *contextmodel.ReqContext) {
+		if !c.IsSignedIn {
+			notAuthorized(c)
+			return
+		}
+
+		pluginID := web.Params(c.Req)[":id"]
+		p, exists := ps.Plugin(c.Req.Context(), pluginID)
+		if !exists {
+			c.JsonApiErr(http.StatusNotFound, "Plugin not found", nil)
+			return
+		}
+
+		found := false
+		allowed := false
+		path := normalizeIncludePath(c.Req.URL.Path)
+
+		hasAccess := ac.HasAccess(accessControl, c)
+		for _, i := range p.Includes {
+			if i.Type != "page" {
+				continue
+			}
+
+			if normalizeIncludePath(i.Path) == path {
```

Comment:
Does (and should? :sweat_smile:) this work when route are defined like this: https://github.com/grafana/oncall/blob/863af25994d766719f4c488cdc24671b4a86ec4e/grafana-plugin/src/plugin.json#L60

Reply:
@grafana/grafana-oncall-backend Do you require the ability to define different access control per query param (IE `tab` in this case)? Otherwise this will just fallback to using the path